File: app/src/main/java/com/github/sewerina/meter_readings/ui/MessageServiceModule.java
package com.github.sewerina.meter_readings.ui;

import android.app.Application;

import dagger.Module;
import dagger.Provides;

@Module(includes = AppContextModule.class)
public class MessageServiceModule {

    @Provides
    public MessageService messageService(Application appContext) {
        return new MessageService(appContext);
    }
}

File: app/src/main/java/com/github/sewerina/meter_readings/di/MessageServiceModule.kt
package com.github.sewerina.meter_readings.di

import android.app.Application
import com.github.sewerina.meter_readings.ui.MessageService
import dagger.Module
import dagger.Provides

@Module(includes = [AppContextModule::class])
class MessageServiceModule {
    @Provides
    fun messageService(appContext: Application): MessageService {
        return MessageService(appContext)
    }
}